plugins/async-node/core/src/index.test.ts

```diff
@@ -340,3 +342,29 @@ test('replaces async nodes with chained multiNodes singular', async () => {
   expect(view?.actions[1].asset.type).toBe('text');
   expect(view?.actions[2].asset.type).toBe('text');
 });
+
+test('should call onAsyncNode hook when async node is encountered', async () => {
+  const plugin = new AsyncNodePlugin();
+  // Mock function to check the node passed to the hook
+  const mockFn = jest.fn();
+
+  // Call the tap method of the onAsyncNode hook
+  plugin.hooks.onAsyncNode.tap('test', async (node: Node.Async) => {
+    mockFn(node);
```

Comment:
is this needed? are we not able to grab the arg here and verify that it matches up to [this](https://github.com/player-ui/player/pull/371/files#diff-fbc47a3e301bdf595a6d686ded538fe6767e534af8821c421f97dc56264d1332R23-R25)

Reply:
Yes this is needed because Inside the function, the first line is a call to mockFn with the node as an argument. mockFn is a mock function, typically used to track calls to the function and the arguments it was called with. Im using this mock function in line no 364 and 365 to verify that the actual response received as in when the async node has been identified if it matches with the mock response or not. This how we can actually verify the response received